core/validation/failures.py

```diff
@@ -230,8 +230,21 @@ def __str__(self):
             f"is non-nullable but contains a null value(s)."
         )
 
-    def to_user_centered_components(self) -> tuple[str, str, str, str]:
-        return "Unimplemented", "Unimplemented", "Unimplemented", "Unimplemented"
+    def to_user_centered_components(self) -> tuple[str, str, str]:
+        sheet = INTERNAL_TABLE_TO_FORM_TAB[self.sheet]
+        column, section = INTERNAL_COLUMN_TO_FORM_COLUMN_AND_SECTION[self.column]
+
+        message = (
+            f'There are blank cells in column(s): "{column}". '
+            f"Use the space provided to tell us the relevant information"
+        )
+
+        # additional logic if Project Outputs as always has same section
+        # and can conflict with Outcomes for Unit of Measurement
+        if sheet == "Project Outputs":
+            section = "Project Outputs"
+
+        return sheet, section, message
```

Comment:
What are the individual cases within the submission that you expect this to appear for?

Reply:
this is mostly for my understanding

Reply:
A lot of the parent tables are populated by information generated at the point of ingestion, and others will be caught by pre-transformation checks. The questions in e.g. Programme Progress are not filled out by the user, so even though they're non-nullable we wouldn't expect them to be caught by this. Outcomes, Outputs, and RiskRegister won't fall under this as the first two have a dropdown that populates Unit of Measurement (but which if not dropped down will cause UoM to be null, but that's a temporary issue), and the latter won't be extracted if Risk Name (the only non-nullable field) is not present.

That leaves the answers in Place Details, various fields in Project Details, a couple of fields in Funding, and potentially Private Investments (although I'm unsure about the latter as have no experience ingesting rounds where it's present).

Reply:
- Place Details won't evoke this failure because Answer is nullable in the schema
- Funding doesn't have any evoke it either
- Private Investment doesn't evoke either because neither of the user fields are non-nullable in the schema

It sounds like this will change for Round 4, and we'll have to make the validation tighter. These changes should come in subsequent tickets IMO though as they've only just been loosely defined.

I can only think of Location (and that weird UoM issue that we've noted) that actually evokes this...

Just wanted to verify it with your understanding as well

Reply:
If it's only Location in Project Details that will be evoked by this then that makes testing easier. I was labouring under a misapprehension because this issue occurred for almost every sheet for historical rounds.

Reply:
Cool - we'll definitely have more to add cases for Round 4. We had to keep loosening the validation for Round 3 U seem to remember because they had edge cases where they didn't want to submit data for certain fields so we gave them an ultimatum and they decided to remove the validation. 

I think add a test for the UoM as well as Location and then it should change when we make the change to instead raise a dropdown error when that ticket is actioned.